### PR TITLE
Add UCX patch to allow overriding modules

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.4
See https://gist.github.com/b76eab31eb5f55d3736899aa03777cd3 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/3f30189ea7a5dd3b805fe0de3644554c for a full test report.

File: easybuild/easyconfigs/u/UCX/UCX-1.10.0-GCCcore-10.3.0.eb [hunk 1]
```diff
@@ -28,6 +32,7 @@ builddependencies = [
 osdependencies = [OS_PKG_IBVERBS_DEV]
 
 dependencies = [
+    ('zlib', '1.2.11'),
```
Comment (verdurin) [inline easybuild/easyconfigs/u/UCX/UCX-1.10.0-GCCcore-10.3.0.eb:35]: @Micket Any comment on adding this dependency, which doesn't seem to be related to the main purpose of the PR?

Comment (Micket) [reply easybuild/easyconfigs/u/UCX/UCX-1.10.0-GCCcore-10.3.0.eb:35]: I just noticed we have missed it adding it. `ldd  $EBROOTUCX/lib/libucs.so`

**Review by verdurin — APPROVED** [review]
Looks fine.

**Comment by verdurin** [comment]
Going in, thanks @Micket!

File: easybuild/easyconfigs/u/UCX/UCX-1.10.0-dynamic_modules.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# UCX is hardcoded to only load the modules it is compiled with.
+# This patch adds an optional EB_xxx_MODULES that can override this list, allowing new modules to be loaded later.
```
Comment (boegel) [inline easybuild/easyconfigs/u/UCX/UCX-1.10.0-dynamic_modules.patch:2]: @Micket Comment says `EB_xxx`, while it's actually `EC_UCX_xxx`, we should fix that to avoid confusion?

Comment (boegel) [reply easybuild/easyconfigs/u/UCX/UCX-1.10.0-dynamic_modules.patch:2]: fixed in #12984
